File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        write_idx = 0

        for i in range(0, len(nums)):
            if nums[i] != val:
                nums[write_idx] = nums[i]
                write_idx += 1
            
        return write_idx

File: Data Structures & Algorithms/maximum-sum-circular-subarray/submission-0.cpp
class Solution {
public:
    int maxSubarraySumCircular(vector<int>& nums) {
        int runningMax = 0, runningMin = 0;
        int maxMiddleSum = nums[0], minMiddleSum = nums[0];
        int arraySum = 0;

        for (int num : nums) {
            arraySum += num;

            runningMax = max(num, runningMax+num);
            maxMiddleSum = max(maxMiddleSum, runningMax);

            runningMin = min(num, runningMin + num);
            minMiddleSum = min(minMiddleSum, runningMin);
        }

        if (maxMiddleSum < 0) {
            return maxMiddleSum;
        }

        return max(maxMiddleSum, arraySum - minMiddleSum);
    }
};